config/jobs/kubernetes/sig-instrumentation/sig-instrumentation-kind-periodics.yaml

```diff
@@ -46,6 +46,8 @@ periodics:
       # Options from https://github.com/kubernetes-sigs/kind/blob/d1eecc46e30cac9d35cd32dc52677ef75ec22e18/hack/ci/e2e-k8s.sh#L79-L83
       - name: CLUSTER_LOG_FORMAT
         value: json
+      - name: GOMAXPROCS
```

Comment:
@dims did some work to auto-detect this, so I'm not 100% sure its needed now.  Maybe we could remove it from the existing canaries and let it run over the weekend to verify?

Reply:
@tzneal We have to revisit this because it's not working for all jobs. For example, `ci-benchmark-scheduler-perf-master-eks-canary` started passing only after we added `GOMAXPROCS` in #29423 today. I'll take a look into this sometimes next week when I have time, but I think we shouldn't block on this.

Reply:
SGTM, I agree on not-blocking.  I wasn't aware of the other issue.

Reply:
We aren't normally setting this on kind jobs, but unit tests in kubernetes auto set it now.

I don't think we should proliferate setting this in the CI jobs, because in the future this will be incorrectly configured and go or kubernetes will be automatically setting the correct value if not already set.

Reply:
> I don't think we should proliferate setting this in the CI jobs, because in the future this will be incorrectly configured and go or kubernetes will be automatically setting the correct value if not already set.

I agree with this. Go is going to fix this, but we need some sort of workaround until then. I'm wondering... triaging where we need to use this mitigation this will take a lot of time. On the other side, having GOMAXPROCS set for some jobs can become an issue for reason Ben stated. What if we create a very simple webhook to set GOMAXPROCS if it's not set? It could just take the value of CPU limit as GOMAXPROCS.

Reply:
Jobs that use GOMAXPROCS are dropped from this batch. I'll think about how we can approach those jobs in a better way.